### PR TITLE
Change Docker setup to match infra, add deployment pipeline

### PR DESCRIPTION
**Pull request by badrange**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    DOCKER_BUILD_ARG_REACT_APP_OIDC_REDIRECT_URI: "https://kukkuu-ui.test.kuva.hel.ninja/callback" # FIXME (see above)
+    DOCKER_BUILD_ARG_REACT_APP_PROFILE_GRAPHQL: "https://kukkuu.test.hel.ninja/graphql"
+  only:
+    refs:
+      - master
+
+review:
+  variables:
+    DOCKER_IMAGE_NAME: "$CI_PROJECT_NAME-review"
+    POSTGRES_ENABLED: 0 #What is this for? This is a react project?
```
Comment (Zeukkari) [inline .gitlab-ci.yml]: There's a database set up by default so explicitly disabling it is necessary.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  DOCKER_BUILD_ARG_REACT_APP_OIDC_CLIENT_ID: "https://api.hel.fi/auth/kukkuu-ui"
+
+review:
+  # These variables are available only for review env and are merged with the general variables defined above.
+  extends: .build
+  variables:
+    DOCKER_IMAGE_NAME: "$CI_PROJECT_NAME-review"
+    DOCKER_BUILD_ARG_REACT_APP_OIDC_AUTHORITY: "https://tunnistamo.test.kuva.hel.ninja"
+    DOCKER_BUILD_ARG_REACT_APP_OIDC_REDIRECT_URI: "https://kukkuu-ui.test.kuva.hel.ninja/callback" # FIXME This is a reference to the UI environment itself, is there a gitlab ci variable for this? 
+    DOCKER_BUILD_ARG_REACT_APP_PROFILE_GRAPHQL: "https://kukkuu.test.hel.ninja/graphql"
```
Comment (Zeukkari) [inline .gitlab-ci.yml]: Rename `DOCKER_BUILD_ARG_REACT_APP_API_URI` -> `DOCKER_BUILD_ARG_REACT_APP_API_URI`

OR

rename `REACT_APP_API_URI` in Dockerfile and client.ts to `REACT_APP_PROFILE_GRAPHQL`

File: .gitlab-ci.yml (hunk 1 above)
Comment (badrange) [reply .gitlab-ci.yml]: Aha! Good to know.

File: .gitlab-ci.yml (hunk 2 above)
Comment (badrange) [reply .gitlab-ci.yml]: Oh, 🤦🏻‍♂️thank you!

File: Dockerfile [hunk 3]
```diff
@@ -1,13 +1,71 @@
...
+COPY --chown=appuser:appuser . .
+
+# Bake package.json start command into the image
+CMD ["react-scripts", "start"]
+
+# ===================================
+FROM appbase as staticbuilder
+# ===================================
+
+ARG REACT_APP_API_URI
```
Comment (antti-mikael) [inline Dockerfile]: All the given build arguments (`REACT_APP_*`) need to be declared here for them to be usable in `yarn build`.

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
+include:
+  - project: 'city-of-helsinki/kuva/ci-cd-config/ci-configuration'
+    ref: v1
+    file: '/.gitlab-ci-template.yml'
+
+variables:
+  SERVICE_PORT: 80
+  DOCKER_BUILD_ARG_REACT_APP_OIDC_CLIENT_ID: "https://api.hel.fi/auth/kukkuu-ui"
```
Comment (antti-mikael) [inline .gitlab-ci.yml]: I would move this to build jobs' variables even if that means repeating the same value three times.
But that's just a matter of taste.

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+review:
```
[the remark this answers is not in the document]
Comment (badrange) [reply .gitlab-ci.yml]: Thanks, updated.

File: .gitlab-ci.yml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  # These variables are available only for review env and are merged with the general variables defined above.
+  extends: .build
+  variables:
+    DOCKER_IMAGE_NAME: "$CI_PROJECT_NAME-review"
+    DOCKER_BUILD_ARG_REACT_APP_OIDC_AUTHORITY: "https://tunnistamo.test.kuva.hel.ninja"
+    DOCKER_BUILD_ARG_REACT_APP_OIDC_REDIRECT_URI: "https://kukkuu-ui.test.kuva.hel.ninja/callback" # FIXME This is a reference to the UI environment itself, is there a gitlab ci variable for this?
```
[the remark this answers is not in the document]
Comment (badrange) [reply .gitlab-ci.yml]: Turns out that we don't actually need this env var, it isn't used in our project.

File: .gitlab-ci.yml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  variables:
+    DOCKER_IMAGE_NAME: "$CI_PROJECT_NAME-staging"
+    DOCKER_BUILD_ARG_REACT_APP_OIDC_AUTHORITY: "https://tunnistamo.test.kuva.hel.ninja"
+    DOCKER_BUILD_ARG_REACT_APP_OIDC_REDIRECT_URI: "https://kukkuu-ui.test.kuva.hel.ninja/callback" # FIXME (see above)
+    DOCKER_BUILD_ARG_REACT_APP_API_URI: "https://kukkuu.test.hel.ninja/graphql"
+  only:
+    refs:
+      - develop
+
+production:
```
[the remark this answers is not in the document]
Comment (badrange) [reply .gitlab-ci.yml]: Thanks, updated.

File: .gitlab-ci.yml (hunk 4 above)
Comment (badrange) [reply .gitlab-ci.yml]: I realised that we don't need this as a changeable environment variable, so I just removed it. 

Will keep your suggestion in mind (and only not consider it if the list of variables is long..)

File: Dockerfile (hunk 3 above)
Comment (badrange) [reply Dockerfile]: Like this?

**Pull request merged**